src/stripeActions/process-data/index.js

```diff
@@ -174,99 +77,182 @@ export const processData = async (payload) => {
+          hashKey: plans[i].hashKey,
+          apiVersion: plans[i].apiVersion,
+        };
+        const id =
+          plans[i].variant === "variant_b"
+            ? `dxpstudio-pricing-${plans[i][index]._key}-${plans[i][
+                index
+              ].planType.replace(/ /g, "-")}-oneTimePrice-${
+                plans[i][index].price
+              }`
+            : `dxpstudio-pricing-${plans[i][index]._key}-${plans[i][
+                index
+              ].planType.replace(/ /g, "-")}-recurring-monthlyPrice-${
+                plans[i][index].monthlyPrice
+              }-yearlyPrice-${plans[i][index].yearlyPrice}`;
 
-          try {
-            // Check if Product is existed in Stripe
-            const checkProductStatus = await getProductById(productPayload);
+        const checkProductPayload = {
+          credentials,
+          StripeParams: {
+            id,
+          },
+        };
 
-            if (checkProductStatus.meta.status === 200) {
-              // Update Product in stripe
-              // Send the Request to API for Updating Our Product
-              try {
-                const responseUpdateProduct = await updateProductById(
-                  productPayload
-                );
-                // If Success it will return status = 200 and the data which we will be using is ID to update the price
-                if (responseUpdateProduct.meta.status !== 200) {
-                  return {
-                    status: 500,
-                    statusText: `Product Not Updated`,
-                  };
-                }
-              } catch (error) {
-                console.log(error);
-              }
-            } else {
-              // Create Product if not existed in stripe
-              // Send the Request to API for Creating Our Product
-              try {
-                const responseCreateProduct = await createProduct(
-                  productPayload
-                );
+        try {
+          const checkProductURL = `${
+            SANITY_STUDIO_APP_URL || "https://dxpstudio.webriq.com"
+          }/api/payments/stripe?resource=products&action=retrieve`;
+          const response = await fetch(checkProductURL, {
+            headers: {
+              "Content-Type": "application/json",
+            },
+            method: "POST",
+            body: JSON.stringify(checkProductPayload),
+          });
+          const checkProductStatus = await response.json();
+          if (checkProductStatus.id) {
+            const updateProductPayload = {
+              credentials,
+              StripeParams: {
+                id,
+                params: {
+                  metadata: !plans[i][index].planIncludes
+                    ? {}
+                    : plans[i][index].planIncludes,
+                  name: plans[i][index].planType,
+                  description: plans[i][index].description,
+                },
+              },
+            };
```

Comment:
This payload needs to be updated as per https://github.com/webriq-pagebuilder/app/pull/63#discussion_r719403192